Add documentation for jwt_clock_skew_tolerance_seconds setting that resolves authentication errors (#3251)
opensearch-project/documentation-website: _security/authentication-backends/saml.md

`idp.entity_id` | The entity ID of your IdP. Required.
`sp.entity_id` | The entity ID of the service provider. Required.

> ## Time disparity compensation for JSON Web Token validation

Reviewer: I just re-checked SAML and for the `nbf` claim it is using the OpenSearch node's clock to determine this claim: https://github.com/opensearch-project/security/blob/main/src/main/java/com/amazon/dlic/auth/http/saml/AuthTokenProcessorHandler.java#L294-L295

For expiration it is getting the value from the SAML response after successful authentication. I think this is good to leave in here, but SAML is an example where the `nbf` come's from the OpenSearch node's clock and the `exp` comes from the authentication server.

> ## Time disparity compensation for JSON Web Token validation

Reviewer 2: ```suggestion
## Time disparity compensation for JWT validation
```

> Occasionally you may find that the clock times between the authentication server and the OpenSearch node are not perfectly synchronized. When this is the case, even by a few seconds, the system that either issues or receives a JWT may try to validate `nbf` (not before) and `exp` (expiration) claims and fail to authenticate the user due to the time disparity.

Reviewer 2: Apply preceding edits to the duplicate content.

## Exchange key settings

> SAML, unlike other protocols, is not meant to be used for exchanging user credentials with each request. The security plugin trades the SAML response for a lightweight JSON Web Token that stores the validated user attributes. This token is signed by an exchange key that you can choose freely. Note that when you change this key, all tokens signed with it become invalid immediately.

Reviewer 2: ```suggestion
SAML, unlike other protocols, is not meant to be used for exchanging user credentials with each request. The Security plugin trades the SAML response for a lightweight JWT that stores the validated user attributes. This token is signed by an exchange key that you can choose freely. Note that when you change this key, all tokens signed with it become invalid immediately.
```

> SAML, unlike other protocols, is not meant to be used for exchanging user credentials with each request. The security plugin trades the SAML response for a lightweight JSON Web Token that stores the validated user attributes. This token is signed by an exchange key that you can choose freely. Note that when you change this key, all tokens signed with it become invalid immediately.

Reviewer 2: Instead of "that you can choose freely", "of your choice"?

> Name | Description

Reviewer 3: To prevent any table rendering issues, I would add a pipe before "Name" and after "Description", and the justification on the next line.

> Name | Description

Reviewer 3: ```suggestion
Name | description
```
